pkg/server/smb_server.go

```diff
@@ -46,16 +46,26 @@ func (h *SMBServer) ListenAndServe(smbAlive chan bool) error {
 	defer func() {
 		smbAlive <- false
 	}()
-	tmpFile, err := os.CreateTemp("", "")
+
+	var err error
+	h.tmpFile, err = fileutil.GetTempFileName()
+	if err != nil {
+		return err
+	}
+
+	pyFileName, err := fileutil.GetTempFileName()
 	if err != nil {
 		return err
 	}
-	h.tmpFile = tmpFile.Name()
-	tmpFile.Close()
-	// execute smb_server.py - only works with ./interactsh-server
-	cmdLine := fmt.Sprintf("python3 smb_server.py %s %d", h.tmpFile, h.options.SmbPort)
-	args := strings.Fields(cmdLine)
-	h.cmd = exec.Command(args[0], args[1:]...)
+	pyFileName += ".py"
+
+	if err := os.WriteFile(pyFileName, []byte(pySmbServer), os.ModePerm); err != nil {
```

Comment:
We can consider using `permissionutil.TempFilePermission`.

Reply:
opted for this as we need also `+x`